Use `PACKAGE_NAME` in messages rather than hardcoding "Bitcoin Core"

hebasto opened the pull request:
Usually, we do not hardcode "Bitcoin Core" in the user-faced messages.

See:
- bitcoin/bitcoin#18646
- bitcoin/bitcoin#19282

Also grammar has been improved -- singular instead of plural.

hebasto committed:
Use `PACKAGE_NAME` in messages rather than hardcoding "Bitcoin Core"

jarolrod reviewed (commented):
This doesn't work as is
<img width="162" alt="Screen Shot 2022-09-20 at 11 20 00 PM" src="https://user-images.githubusercontent.com/23396902/191407015-a0d903a5-265b-4ab8-b5af-96ae5ae584ee.png">

hebasto commented:
@jarolrod 
> This doesn't work as is

Thanks! Updated.

luke-jr commented on src/init.cpp:
```diff
@@ -1662,11 +1662,12 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     const auto BadPortWarning = [](const char* prefix, uint16_t port) {
-        return strprintf(_("%s request to listen on port %u. This port is considered \"bad\" and "
-                           "thus it is unlikely that any Bitcoin Core peers connect to it. See "
+        return strprintf(_("%s requests to listen on port %u. This port is considered \"bad\" and "
```
`-bind` is more of an action than a noun, so I think the original here is correct

hebasto replied:
Thanks. A commit [dropped](https://github.com/bitcoin/bitcoin/pull/26142#issuecomment-1260104579).

luke-jr commented on src/init.cpp:
```diff
@@ -1662,11 +1662,12 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
-        return strprintf(_("%s request to listen on port %u. This port is considered \"bad\" and "
-                           "thus it is unlikely that any Bitcoin Core peers connect to it. See "
+        return strprintf(_("%s requests to listen on port %u. This port is considered \"bad\" and "
+                           "thus it is unlikely that any %s peers connect to it. See "
```
```suggestion
                           "thus it is unlikely that any peers will connect to it. See "
```

No reason to name the software specifically here?

jarolrod replied:
true

jarolrod replied:
On second thought, we are touching this, but it isn’t a part of the pr’s goal to address something like this; seems appropriate for a follow-up if someone wants to suggest this change.

hebasto replied:
> No reason to name the software specifically here?

[Updated](https://github.com/bitcoin/bitcoin/pull/26142#issuecomment-1260104579).

luke-jr commented on src/qt/forms/optionsdialog.ui:
```diff
@@ -286,7 +286,7 @@
              <widget class="QLineEdit" name="externalSignerPath">
               <property name="toolTip">
-               <string>Full path to a Bitcoin Core compatible script (e.g. C:\Downloads\hwi.exe or /Users/you/Downloads/hwi.py). Beware: malware can steal your coins!</string>
+               <string>Full path to a %1 compatible script (e.g. C:\Downloads\hwi.exe or /Users/you/Downloads/hwi.py). Beware: malware can steal your coins!</string>
```
ACK

luke-jr commented on src/qt/optionsdialog.cpp:
```diff
@@ -91,7 +91,9 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
 
-#ifndef ENABLE_EXTERNAL_SIGNER
+#ifdef ENABLE_EXTERNAL_SIGNER
+    ui->externalSignerPath->setToolTip(ui->externalSignerPath->toolTip().arg(PACKAGE_NAME));
```
Might be nice if there was a BIP we could specify here...

hebasto commented:
Updated d895eb254b8fee30e13c190409f7b53a447c6416 -> b147322a7a387423164de5a7d91a33eacad51689 ([pr26142.02](https://github.com/hebasto/bitcoin/commits/pr26142.02) -> [pr26142.03](https://github.com/hebasto/bitcoin/commits/pr26142.03), [diff](https://github.com/hebasto/bitcoin/compare/pr26142.02..pr26142.03)):

- addressed @luke-jr's comments

luke-jr reviewed (approved):
utACK

hebasto commented:
Friendly ping @jarolrod.

jarolrod reviewed (approved):
ACK b147322a7a387423164de5a7d91a33eacad51689

No visual difference between pr and master, as expected.

| pr | master |
| -- | ------ |
| <img width="180" alt="Screen Shot 2022-10-19 at 2 39 44 AM" src="https://user-images.githubusercontent.com/23396902/196618670-1d6a0fea-fcae-4a22-b1e0-b518fbe981a9.png"> | <img width="180" alt="Screen Shot 2022-10-19 at 2 55 41 AM" src="https://user-images.githubusercontent.com/23396902/196618778-1987f465-12b0-4716-8eef-d1c28f7fa721.png"> |
